code/game/objects/items/tanks/jetpack.dm

```diff
@@ -119,16 +119,14 @@
 	name = "improvised jetpack"
 	desc = "A jetpack made from two air tanks, a fire extinguisher and some atmospherics equipment. It doesn't look like it can hold much."
 	icon_state = "jetpack-improvised"
-	inhand_icon_state = "jetpack-sec"
+	inhand_icon_state = "jetpack-improvised"
 	worn_icon = null
-	worn_icon_state = "jetpack-sec"
+	worn_icon_state = "jetpack-improvised"
 	volume = 20 //normal jetpacks have 70 volume
 	gas_type = null //it starts empty
 	full_speed = FALSE //moves at hardsuit jetpack speeds
 
 /obj/item/tank/jetpack/improvised/allow_thrust(num, mob/living/user)
-	if(!on)
-		return
 	if((num < 0.005 || air_contents.total_moles() < num))
 		turn_off(user)
 		return
```

Comment:
The issue with this proc is less that it still has the old check for on, but more that it's a full copypasta of `/obj/item/tank/jetpack/proc/allow_thrust` with one difference to allow for random failure. It should just check for the random failure, and if it doesn't go through, call parent

Reply:
Code has been changed accordingly 👍